### PR TITLE
feat: add "tsserver.trace" init option for tracing tsserver

### PR DESCRIPTION
**Pull request by rchl**
Enables nicely formatted log messages with tsserver communication delivered through the LSP messages (so visible in editor's log of LSP communication).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,20 @@ The language server accepts various settings through the `initializationOptions`
 | locale            | string   | The locale to use to show error messages. |
 | plugins           | object[] | An array of `{ name: string, location: string }` objects for registering a Typescript plugins. **Default**: []                                                                                                                                                         |
 | preferences       | object   | Preferences passed to the Typescript (`tsserver`) process. See below for more info.                                                                                                                              |
+| tsserver          | object   | Options related to the `tsserver` process. See below for more info.                                                                                                                              |
+
+The `tsserver` setting specifies additional options related to the internal `tsserver` process, like tracing and logging.
+
+```ts
+interface TsserverOptions {
```
Comment (predragnikolic) [inline README.md:91]: Maybe rename `TsserverOptions` to `TsServerOptions`?

File: README.md [hunk 2]
```diff
@@ -83,6 +83,20 @@ The language server accepts various settings through the `initializationOptions`
 | locale            | string   | The locale to use to show error messages. |
 | plugins           | object[] | An array of `{ name: string, location: string }` objects for registering a Typescript plugins. **Default**: []                                                                                                                                                         |
 | preferences       | object   | Preferences passed to the Typescript (`tsserver`) process. See below for more info.                                                                                                                              |
+| tsserver          | object   | Options related to the `tsserver` process. See below for more info.                                                                                                                              |
```
Comment (predragnikolic) [inline README.md:86]: what is the difference between `initializationOptions.logVerbosity`, `initializationOptions.tsserver.trace` and the "--tsserver-log-verbosity" cli flag?

Comment (rchl) [reply README.md:86]: `initializationOptions.logVerbosity` and `--tsserver-log-verbosity` are related to file-based logs.

I have plans to do further cleanup here, removing some command line options and unifying some naming.

File: README.md (hunk 1 above)
Comment (rchl) [reply README.md:91]: Then to match the property should also be called `tsServer` probably and I wanted to match vscode here.

File: README.md (hunk 2 above)
Comment (predragnikolic) [reply README.md:86]: Also I just realized that there is a difference between the term "log" and "trace". 
Log is just a text log, no way of knowing how you got to the log.
Trace should gives you information to trace something from the time till its requested till the time it is done.

Comment (rchl) [reply README.md:86]: I'm trying to follow VSCode here as closely as I can to have feature and setting parity.

It's not entirely consistent there either.

For example there is a setting `typescript.tsserver.trace` which enables tsserver request logging to the panel and there is `typescript.tsserver.enableTracing` which enables creating performance tracing data to a file...

Comment (rchl) [reply README.md:86]: In any case, do you see something that could be improved here or are you just stating your discoveries because I'm sure  :)

Comment (predragnikolic) [reply README.md:86]: I am just stating my discoveries :⁠-⁠D 

I would resolve the conversation, but sometimes Github doesn't show me the resolve conversation button.

**Pull request merged**